app/views/helpers/AmendmentDetailsHelper.scala

```diff
@@ -226,18 +226,34 @@ class AmendmentDetailsHelper @Inject() (
     def fetchCountry(countryCode: Option[String]): Option[String] =
       countryCode.flatMap(countryHelper.getShortNameForCountryCode)
 
-    def updateRoutingCountry(routingCountry: RoutingCountry): Option[RoutingCountry] =
+    def handleRoutingCountry(routingCountry: RoutingCountry): Option[RoutingCountry] =
       Some(routingCountry.copy(country = Country(fetchCountry(routingCountry.country.code))))
 
-    def handlePotentialCountryCode(maybeCode: String): String =
-      if (maybeCode.isBlank) maybeCode
+    def handleString(maybeCode: String): Option[String] =
+      if (maybeCode.isBlank) Some(maybeCode)
       else if (maybeCode.length == 2 && isValidCountryCode(maybeCode))
```

Comment:
Nit: would it maybe worth to have here something like `else if (maybeCode.trim.length == 2` ? Since you are also testing with `isBlank`